lib/template.html

```diff
@@ -206,28 +206,60 @@
     "headline": {{ seo_page_title | jsonify }},
 {% endif %}
 
+{% if seo_author %}
+    "author": {
+      "@type": "Person",
+      "name": {{ seo_author | jsonify }}
+    },
+{% endif %}
+
 {% if seo_page_image %}
-    "image": {{ seo_page_image | jsonify }},
+    {% if page.image.height && page.image.width %}
+      "image": {
+          "@type": "ImageObject",
+          "url": {{ seo_page_image | jsonify }},
+          "height": {{ page.image.height | jsonify }},
+          "width": {{ page.image.width | jsonify }}
+      },
+    {% else %}
+        "image": {{ seo_page_image | jsonify }},
+    {% endif %}
 {% endif %}
 
 {% if page.date %}
     "datePublished": {{ page.date | date_to_xmlschema | jsonify }},
 {% endif %}
 
+{% if page.dateModified %}
+    "dateModified": {{ page.dateModified | date_to_xmlschema | jsonify }},
+{% elsif page.date %}
+    "dateModified": {{ page.date | date_to_xmlschema | jsonify }},
+{% endif %}
```

Comment:
Sure. [Google's Docs](https://developers.google.com/search/docs/data-types/articles) mention that the `dateModified` field is recommended. I can't give an exact explanation for its use other than it receives Google's recommendation. 

Under this documentation, is listed as recommended for AMP pages, and ignored on non-AMP. But then again, it also says that `datePublished` is ignored on non-AMP pages. What are your thoughts?